Mathlib/Order/ReltoGaloisConnection.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2024 Lagrange Mathematics and Computing Research Center. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anthony Bordg
+-/
+import Mathlib.Data.Rel
+import Mathlib.Order.GaloisConnection
+
+/-!
+# The Galois Connection Induced by a Relation
+
+In this file, we show that an arbitrary relation `R` between a pair of types `α` and `β` defines
+a pair `leftDual` and `rightDual` of adjoint order-preserving maps between the corresponding posets
+`Set α` and `(Set β)ᵒᵈ`.
+We define `lFixedPoints` (resp. `rFixedPoints`) as the set of fixed points `J` (resp. `I`) of
+`Set α` (resp. `(Set β)ᵒᵈ`) such that `rightDual (leftDual J) = J` (resp.
+`leftDual (rightDual I) = I`).
+
+## Main Results
+
+⋆ `galoisConnection`: we prove that the maps `leftDual` and `rightDual` form a Galois connection.
+⋆ `equivFixedPoints`: we prove that the said maps induce inverse bijections between the sets
+of fixed points.
+
+## References
+
+⋆ Engendrement de topologies, démontrabilité et opérations sur les sous-topos, Olivia Caramello and
+Laurent Lafforgue (in preparation)
+
+## Tags
+
+relation, Galois connection, induced bijection, fixed points
+-/
+
+section
+
+universe u v
+
+variable {α : Type u} {β : Type v} (R : Rel α β)
+
+namespace Rel
+
+/-! ### Pairs of adjoint maps defined by relations -/
+
+open OrderDual
+
+/-- `leftDual` maps any set `J` of elements of type `α` to the set `{b : β | ∀ a ∈ J, R a b}` of
+elements `b` of type `β` such that `R a b` for every element `a` of `J`. -/
+def leftDual (J : Set α) : (Set β)ᵒᵈ := toDual {b : β | ∀ a ∈ J, R a b}
+
+/-- `rightDual` maps any set `I` of elements of type `β` to the set `{a : α | ∀ b ∈ ofDual I, R a b}`
+of elements `a` of type `α` such that `R a b` for every element `b` of `I`. -/
+def rightDual (I : (Set β)ᵒᵈ) : Set α := {a : α | ∀ b ∈ ofDual I, R a b}
+
+/-- The pair of functions `leftDual` and `rightDual` forms a Galois connection. -/
+theorem galoisConnection : GaloisConnection (leftDual R) (rightDual R) := by
+    intros J I; apply Iff.trans (b := ∀ b ∈ ofDual I, ∀ a ∈ J, R a b)
+    · constructor
+      · intro h; apply h
+      · intro h; apply h
+    · constructor
+      · intro h; unfold rightDual; intro a ha b hb; apply h; simpa; exact ha
+      · intro h; intro b hb a ha; apply h; simpa; exact hb
```

Comment:
This is not in line with the style guide. Please put at most one tactic per line unless one of the following applies:
* It's a one-line proof
* The tactics are chained with `<;>`

Reply:
Line 62 and line 63 shouldn't be considered one-line proofs, although they close a goal?

Reply:
No, they are part of a proof that's longer than one line. They are one-line *sub*proofs.